src/qz/utils/PrintingUtilities.java

```diff
@@ -59,16 +82,7 @@ public static PrinterResolution getNativeDensity(PrintService service) {
         PrinterResolution pRes = (PrinterResolution)service.getDefaultAttributeValue(PrinterResolution.class);
 
         if (pRes == null && !SystemUtilities.isWindows()) {
-            if (CUPS_DPI == null || !CUPS_DESC.containsValue(service.getName())) {
-                CUPS_DESC = ShellUtilities.getCupsPrinters();
-                CUPS_DPI = ShellUtilities.getCupsDensities(CUPS_DESC);
-            }
-
-            for(String name : CUPS_DESC.keySet()) {
-                if (CUPS_DESC.get(name).equals(service.getName())) {
-                    return CUPS_DPI.get(name);
-                }
-            }
+            return CUPS_DPI.get(getPrinterId(service.getName()));
```

Comment:
> Why was CUPS_DPI moved out if it is only needed in this method?

It certainly doesn't need to be, but since `CUPS_DPI` and `CUPS_DESC` are so closely coupled, it didn't make sense in my mind to cache one now and then the other later, it just seemed like we'd have strange scenarios where one was updated and the other was not.  Thoughts?